pkg/controller/nodes/node_exec_context.go

```diff
@@ -198,19 +203,24 @@ func (c *nodeExecutor) newNodeExecContextDefault(ctx context.Context, currentNod
 		return nil
 	}
 
-	interrutible := executionContext.IsInterruptible()
+	interruptible := executionContext.IsInterruptible()
 	if n.IsInterruptible() != nil {
-		interrutible = *n.IsInterruptible()
+		interruptible = *n.IsInterruptible()
 	}
 
 	s := nl.GetNodeExecutionStatus(ctx, currentNodeID)
 
 	// a node is not considered interruptible if the system failures have exceeded the configured threshold
-	if interrutible && s.GetSystemFailures() >= c.interruptibleFailureThreshold {
-		interrutible = false
+	if interruptible && s.GetSystemFailures() >= c.interruptibleFailureThreshold {
+		interruptible = false
 		c.metrics.InterruptedThresholdHit.Inc(ctx)
 	}
 
+	rawOutputPrefix := c.defaultDataSandbox
+	if executionContext.GetRawOutputDataConfig().RawOutputDataConfig != nil && executionContext.GetRawOutputDataConfig().OutputLocationPrefix != "" {
```

Comment:
hmm why the 2 checks?
So node execution context is recursive, so for a node, it should always be parent node exec context + some stuff?

Reply:
i can remove

Reply:
Removed.  But this meant I had to fix a bunch of unit tests.